azure-tools-common/src/main/java/com/microsoft/azure/common/function/IFunctionContext.java

```diff
@@ -0,0 +1,38 @@
+package com.microsoft.azure.common.function;
+
+import java.util.Map;
+
+import com.microsoft.azure.auth.configuration.AuthConfiguration;
+import com.microsoft.azure.common.docker.IDockerCredentialProvider;
+import com.microsoft.azure.common.project.IProject;
+import com.microsoft.azure.maven.function.configurations.RuntimeConfiguration;
+
+public interface IFunctionContext {
+	String getDeploymentStagingDirectoryPath();
+
+	String getSubscription();
+
+	String getAppName();
+
+	String getResourceGroup();
+
+	RuntimeConfiguration getRuntime();
+
+	String getRegion();
+
+	String getPricingTier();
+
+	String getAppServicePlanResourceGroup();
+
+	String getAppServicePlanName();
+
+	Map getAppSettings();
+
+	AuthConfiguration getAuth();
+
+	String getDeploymentType();
+
+	IDockerCredentialProvider getDockerCredentialProvider();
```

Comment:
`IFunctionContext` should be the abstraction of function project, and plays a role like pojo? So i think it should extends `IDockerCredentialProvider` provide userName and password instead of provide `IDockerCredentialProvider`

Reply:
When os is not docker, this provider will be null.